From 7b43a1812e03f0004e4da2103f24a10fd7c8a304 Mon Sep 17 00:00:00 2001
From: Ondrej Jirman <megi@xff.cz>
Date: Mon, 3 Aug 2026 00:52:35 +0200
Subject: [PATCH] ARM: dts: allwinner: sun8i-a83t-tbs-a711: Gate the speaker
 amplifier

simple-audio-amplifier powers its DRV widget from either of its two
inputs, and only INL was routed. An unrouted DAPM input pin counts as
connected, so DRV was powered from boot and the enable gpio (PG13)
read out-hi with nothing playing.

Route both pins from the one speaker output the board uses, as
sun50i-a64-pinephone.dtsi does for the same amplifier.

Signed-off-by: Ondrej Jirman <megi@xff.cz>
---
 arch/arm/boot/dts/allwinner/sun8i-a83t-tbs-a711.dts | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/arch/arm/boot/dts/allwinner/sun8i-a83t-tbs-a711.dts b/arch/arm/boot/dts/allwinner/sun8i-a83t-tbs-a711.dts
index af0c23889b60..76c2da0715d4 100644
--- a/arch/arm/boot/dts/allwinner/sun8i-a83t-tbs-a711.dts
+++ b/arch/arm/boot/dts/allwinner/sun8i-a83t-tbs-a711.dts
@@ -196,7 +196,9 @@ sound: sound {
 		simple-audio-card,routing =
 				"Headphone Jack", "HP",
 				"Internal Speaker", "Speaker Amp OUTL",
+				"Internal Speaker", "Speaker Amp OUTR",
 				"Speaker Amp INL", "SPKOUTL",
+				"Speaker Amp INR", "SPKOUTL",
 				"Left DAC", "DACL",
 				"Right DAC", "DACR",
 				"ADCL", "Left ADC",
-- 
2.43.0

